src/blog/templates/blog/blog_page.html

```diff
@@ -37,13 +37,13 @@ <h1 class="bold text-4xl mb-0 text-black">
                 {{ page.body|richtext }}
             </article>
 
-            <div class="flex flex-row flex-wrap p-1">
+            <div class="p-4 grid md:grid-cols-2 lg:grid-cols-3 gap-4">
                 {% for item in page.gallery_images.all %}
-                    <div class="flex-col p-1" style="width: 320px">
-                        {% image item.image fill-320x240 %}
-                        <p>{{ item.caption }}</p>
-                    </div>
-                {% endfor %}    
+                <div class="bg-white rounded-xl overflow-hidden drop-shadow group hover:rotate-3 transition-transform">
+                    {% image item.image fill-703x408 %}
+                    <p style="padding: 0.5rem">{{ item.caption }}</p>
```

Comment:
I would try and refrain from using `style` and just use Tailwindcss's `p-2`, which resolves into `0.5rem` by default.